vsutil/func.py

```diff
@@ -142,3 +144,61 @@ def iterate(base: T, function: Callable[[Union[T, R]], R], count: int) -> Union[
     for _ in range(count):
         v = function(v)
     return v
+
+
+# This function is actually implemented as a class.
+# This makes sure that,
+# when it is used as the value of a class-variable,
+# python does not prefix calls to this function with ``self``.
+# It also allows to forward calls to 
+# - `plugin`,
+# - `signature`,
+# - and `return_signature`
```

Comment:
When are `signature` and `return_signature` used by users? I get that they’re part of the VS API and you probably want the return of `function` to be as close to that as possible, but is there a real case of anyone accessing one of these from Python that justifies this layer of indirection (when they could instead call `.resolved` and not rely on us playing the middle man)?

Reply:
nvm, that was answered above